### DISCUSSION TITLE
RFC: useFieldArray `replace` method

### DISCUSSION THREAD

**Discussion by bluebill1049** [react-hook-form/react-hook-form#6357] [RFC]
## Context

Developers will be required to reset a field array with `reset` API, this not only affects a single field array but the entire form, which could lead to unexpected behavior, you can achieve similar behavior with `setValue` with field array `name`. We want to bring a new API `replace` to make API more declarative and easier for developers who just want to update and replace a specific field array.

### Props

- declarative API for reset a single field array
- more performant than `reset` the entire form

### Cons

- increase in package bundle size.

## Proposal

```tsx
const { control } = useForm({
  defaultValues: {
    test: [{ value: 'lorem' }, { value: 'ipsum' }]
  }
});
const { fields, replace } = useFieldArray({ 
  control, 
  name: 'test'
});

const handleFullyReplacement = (): void => {
  // remove old and set fully new values
  replace([{ value: 'dolor' }, { value: 'sit' }, { value: 'amet' }]);
}
```

## PR

https://github.com/react-hook-form/react-hook-form/pull/6304

cc @jorisre @kotarella1110 @barrymay

**Comment by bluebill1049** [comment]
PR is merged.

**Reply by OlehDutchenko** [reply]
Thanks! 👍

**Reply by bluebill1049** [reply]
Thanks to @dutchenkoOleg for your contribution 🙏

**Comment by bluebill1049** [comment]
released and the doc is updated too.

**Reply by MedinaGitHub** [reply]
@bluebill1049  That method doesn't exist on Version 6 =S I'm using 6.15.5 version. But it was added to the legacy doc

<img width="1180" alt="image" src="https://user-images.githubusercontent.com/16387514/217632689-3de81d7c-a3f9-4000-8a79-1129cc903eeb.png">

**Reply by bluebill1049** [reply]
could you send an issue at the documentation repo? let's fix this.
